### PR TITLE
CV Camera Status Plugin

### PR DESCRIPTION
**Pull request by vedarshshah**
Developed an RQT plugin to display the connection status of the stereo and mono cameras.

Opening as draft for now. A few changes will need to be made after #393 is merged in.

### CODE REVIEW

File: landside/catkin_ws/src/gui/src/gui/camera_status_widget.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+
+    def open_conection_log(self):
+        self.log = CameraStatusLog(self.data_updated, self.status_logs)
+        self.log.exec()
+
+    def check_camera_connection(self, camera_type):
+        if camera_type == CameraStatusDataType.MONO:
+            call_connect_camera_service = CallConnectCameraService(camera_type, self.channel)
+        else:
+            call_connect_camera_service = CallConnectCameraService(camera_type)
```
Comment (Brokemia) [inline landside/catkin_ws/src/gui/src/gui/camera_status_widget.py:204]: Is there a way to generalize this? Everything else seems very camera agnostic, so this hard check for MONO kind of sticks out

**Review by Brokemia — COMMENTED** [review]
Mostly looks good, just a few points of feedback

**Comment by Brokemia** [comment]
@vedarshshah Want to make sure you saw this

File: landside/catkin_ws/src/gui/src/gui/camera_status_widget.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+
+        self.status_logs = {}
+        for camera_type in CameraStatusDataType:
+            self.status_logs[camera_type] = []
+
+        self.status_table.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+
+        self.timer = QTimer(self)
+        self.timer.timeout.connect(self.timer_check)
+        self.timer.start(100)
```
[the remark this answers is not in the document]
Comment (vedarshshah) [reply landside/catkin_ws/src/gui/src/gui/camera_status_widget.py:132]: We've used the time of 100ms in all timers in RQT plugins when we need to repeatedly poll to check if certain rosservice(s) are available. See controls_widget.py, launch_widget.py, offboard_widget.py, sensor_widget.py and system_usage_widget.py for similar timers.

[the remark this answers is not in the document]
Comment (Brokemia) [reply landside/catkin_ws/src/gui/src/gui/camera_status_widget.py:132]: Ah, didn't realize it was on a loop, I assumed it was a one-time delay. That's fine then

File: landside/catkin_ws/src/gui/src/gui/camera_status_widget.py [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+            CameraStatusDataType.MONO: self.check_mono_button
+        }
+
+        self.checking = {}
+        for camera_type in CameraStatusDataType:
+            self.checking[camera_type] = False
+
+        self.status_logs = {}
+        for camera_type in CameraStatusDataType:
+            self.status_logs[camera_type] = []
```
[the remark this answers is not in the document]
Comment (vedarshshah) [reply landside/catkin_ws/src/gui/src/gui/camera_status_widget.py]: Changed the first loop that modifies `self.checking` to use `CAMERA_STATUS_CAMERA_TYPES`. The second loop that modifies `self.status_logs` should iterate over all `CameraStatusDataType`s including PING as the widget stores all messages for PING, MONO, and STEREO in their respective status logs.

**Comment by vedarshshah** [comment]
@Brokemia All comments have been addressed. Please take a look and approve.

File: landside/catkin_ws/src/gui/src/gui/camera_status_widget.py (hunk 1 above)
Comment (Brokemia) [reply landside/catkin_ws/src/gui/src/gui/camera_status_widget.py]: Looks okay to me

**Review by Brokemia — APPROVED** [review]
Looks good now

**Pull request merged**